packages/x-date-pickers/package.json

```diff
@@ -103,7 +103,7 @@
     "@types/moment-jalaali": "^0.7.9",
     "@types/prop-types": "^15.7.12",
     "date-fns": "^2.30.0",
-    "date-fns-jalali": "^2.13.0-0",
+    "date-fns-jalali": "^2.30.0-0",
```

Comment:
I had to bump the version to have the exported `date-fns-jalali/locale/types` file so the build would pass.